pkgs/by-name/da/davs2/package.nix

```diff
@@ -0,0 +1,59 @@
+{
+  lib,
+  fetchFromGitHub,
+  gitUpdater,
+  nasm,
+  stdenv,
+  testers,
+  pkgsBuildBuild,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "davs2";
+  version = "1.7";
+
+  src = fetchFromGitHub {
+    owner = "pkuvcl";
+    repo = "davs2";
+    rev = "refs/tags/${finalAttrs.version}";
+    hash = "sha256-SUY3arrVsFecMcbpmQP0+4rtcSRfQc6pzxZDcEuMWPU=";
+  };
+
+  preConfigure = ''
+    # Generate version.h
+    ./version.sh
+    cd build/linux
+    export AS=nasm
+  '';
+
+  configureFlags = lib.optionals (!stdenv.hostPlatform.isStatic) [
+    (lib.enableFeature true "shared")
+    "--system-libdavs2"
+  ];
+
+  outputs = [
+    "out"
+    "lib"
+    "dev"
+  ];
+
+  nativeBuildInputs = [
+    nasm
+    pkgsBuildBuild.binutils-unwrapped
```

Comment:
Generating code for the build platform?!

Reply:
It's only required in configure: https://github.com/pkuvcl/davs2/blob/b41cf117452e2d73d827f02d3e30aa20f1c721ac/build/linux/configure#L926

Reply:
I'd assume the configure script to expect it to generate code for the hostPlatform though, right?

Reply:
Well no. It only uses the `strings` utility:
```
    if (${cross_prefix}strings -a conftest.o | grep -q BIGE) && (${cross_prefix}strings -a conftest.o | grep -q FPendian) ; then
        define WORDS_BIGENDIAN
        CPU_ENDIAN="big-endian"
    elif !(${cross_prefix}strings -a conftest.o | grep -q EGIB && ${cross_prefix}strings -a conftest.o | grep -q naidnePF) ; then
        die "endian test failed"
    fi
```
In fact it may be easier to just substitute the proper path to strings there.

Reply:
Given that it says `${cross_prefix}` explicitly, it should really just be normal binutils?

Reply:
Yes, I think `strings -a` does not depend on endianness or anything else really.

Reply:
But if we’re doing cross then `pkgsBuildBuild.binutils-wrapped` will not have the right prefix and it will fail, right?

I do not see why we need to bring in the more complicated package sets here rather than just doing what the build system expects.

Reply:
> will not have the right prefix and it will fail, right?

Well cross_prefix is always "", even for cross builds. Perhaps we should get that fixed first.

Besides that I think davs2 is only compatible with x86, unless I missed something looking through the source. (there is only https://github.com/pkuvcl/davs2/tree/master/source/common/x86, no other platforms)

Reply:
But `strings` is in the standard environment. i don’t understand why this dependency is necessary at all.

Reply:
I think I figured it out. See https://github.com/NixOS/nixpkgs/pull/337010/files#r1823161102

Reply:
Removed it.